trigger/build.go

```diff
@@ -62,13 +62,13 @@ func (t *buildTriggers) Fetch(ctx context.Context, e *event.EventLogEntry) error
 
 	var tasks []task.Task
 	if e.EventType == event.BuildGithubCheckFinished {
-		query := db.Query(task.ByBuildIdAndGithubChecks(t.build.Id)).WithFields(task.StatusKey, task.DependsOnKey)
```

Comment:
This looks good, but is there any way we can test it to see if it fixes the bug?

Reply:
<img width="645" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/26798134/bcd43bcc-b072-4ee8-a18f-5a56f9dfe992">

(I'll add unscheduled to statuses we skip)